### PR TITLE
Return 404 on invalid theme type

### LINKED ISSUE #32798: Custom theme and not existing image: error 500 (No enum constant org.keycloak.theme.Theme.Type.IMG)
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

login/ui

### Describe the bug

I'm developing a custom theme in `/opt/keycloak/themes` using a Docker container based on the image `quay.io/keycloak/keycloak:25.0.4`.
In the CSS I mispelled the name of an image file.
I'd expected that Keycloak returned a 404 error (Not Found): it instead returns a 500 error (Internal Server Error).

Indeed, when browsing directly to the URL of the wrong image file (`http://localhost:8080/resources/zewye/img/pages/mispelled-file-name.png`), in the container logs window I see the following:

```
2024-09-11 12:06:39 2024-09-11 10:06:39,647 WARN  [org.keycloak.services] (executor-thread-26) KC-SERVICES0075: Failed to get theme request: java.lang.IllegalArgumentException: No enum constant org.keycloak.theme.Theme.Type.IMG
2024-09-11 12:06:39 at java.base/java.lang.Enum.valueOf(Enum.java:293)
2024-09-11 12:06:39 at org.keycloak.theme.Theme$Type.valueOf(Theme.java:35)
2024-09-11 12:06:39 at org.keycloak.services.resources.ThemeResource.getResource(ThemeResource.java:84)
2024-09-11 12:06:39 at org.keycloak.services.resources.ThemeResource$quarkusrestinvoker$getResource_808824b51c03e102b8d400a19f17c54968201e8a.invoke(Unknown Source)
2024-09-11 12:06:39 at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
2024-09-11 12:06:39 at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
2024-09-11 12:06:39 at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
2024-09-11 12:06:39 at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:582)
2024-09-11 12:06:39 at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
2024-09-11 12:06:39 at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
2024-09-11 12:06:39 at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
2024-09-11 12:06:39 at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
2024-09-11 12:06:39 at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2024-09-11 12:06:39 at java.base/java.lang.Thread.run(Thread.java:1583)
2024-09-11 12:06:39 
2024-09-11 12:06:39 2024-09-11 10:06:39,647 ERROR [org.keycloak.headers.DefaultSecurityHeadersProvider] (executor-thread-26) MediaType not set on path /resources/zewye/img/pages/mispelled-file-name.png, with response status 500
2024-09-11 12:06:39 2024-09-11 10:06:39,648 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-26) Uncaught server error: jakarta.ws.rs.InternalServerErrorException: HTTP 500 Internal Server Error
2024-09-11 12:06:39 at org.keycloak.headers.DefaultSecurityHeadersProvider.addHeaders(DefaultSecurityHeadersProvider.java:75)
2024-09-11 12:06:39 at org.keycloak.services.filters.KeycloakSecurityHeadersFilter.filter(KeycloakSecurityHeadersFilter.java:45)
2024-09-11 12:06:39 at org.jboss.resteasy.reactive.server.handlers.ResourceResponseFilterHandler.handle(ResourceResponseFilterHandler.java:25)
2024-09-11 12:06:39 at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:150)
2024-09-11 12:06:39 at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
2024-09-11 12:06:39 at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:582)
2024-09-11 12:06:39 at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
2024-09-11 12:06:39 at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
2024-09-11 12:06:39 at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
2024-09-11 12:06:39 at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
2024-09-11 12:06:39 at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2024-09-11 12:06:39 at java.base/java.lang.Thread.run(Thread.java:1583)
```

### Version

25.0.4

### Regression

- [ ] The issue is a regression

### Expected behavior

Browsing to a non existinging theme image file should return a 404 (Not Found) error.

### Actual behavior

Browsing to a non existinging theme image file should returns a 500 (Internal Server Error) error.

### How to Reproduce?

1. Create a custom theme
2. In the theme CSS file, add a reference to a non existing URL, for example:
   ```css
   .authentication-wrapper.authentication-basic .authentication-inner:before {
      width: 21rem;
      height: 21rem;
      content: " ";
      position: fixed;
      top: -100px;
      background: url("../../../img/pages/mispelled-file-name.png");
      background-size: contain;
   }
   ```

### Anything else?

_No response_
```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by rmartinc**
Closes #32798

Little fix for the issue. It's just checking the enum for theme type is valid.

### CODE REVIEW

File: services/src/main/java/org/keycloak/services/resources/ThemeResource.java [hunk 1]
```diff
@@ -75,13 +75,14 @@ public class ThemeResource {
     @GET
     @Path("/{version}/{themeType}/{themeName}/{path:.*}")
     public Response getResource(@PathParam("version") String version, @PathParam("themeType") String themType, @PathParam("themeName") String themeName, @PathParam("path") String path) {
-        if (!version.equals(Version.RESOURCES_VERSION)) {
+        final Theme.Type type = getThemeType(themType);
```
Comment (jonkoops) [inline services/src/main/java/org/keycloak/services/resources/ThemeResource.java:78]: Perhaps we can fix this typo while we're at it?
```suggestion
        final Theme.Type type = getThemeType(themeType);
```

File: services/src/main/java/org/keycloak/services/resources/ThemeResource.java [hunk 2]
```diff
@@ -153,6 +154,14 @@ public Response getLocalizationTexts(@PathParam("realm") String realmName, @Quer
 
         return Cors.builder().allowedOrigins("*").auth().add(Response.ok(result));
     }
+
+    private static Theme.Type getThemeType(String themeType) {
+        try {
+            return Theme.Type.valueOf(themeType.toUpperCase());
+        } catch (IllegalArgumentException iae) {
+            return null;
```
Comment (jonkoops) [inline services/src/main/java/org/keycloak/services/resources/ThemeResource.java:162]: Seems like a good place to use an optional, but this works fine too.

**Review by jonkoops — DISMISSED** [review]
LGTM

**Review by mposolda — DISMISSED** [review]
@rmartinc @jonkoops Thanks for the fix and the review!

File: services/src/main/java/org/keycloak/services/resources/ThemeResource.java (hunk 1 above)
Comment (rmartinc) [reply services/src/main/java/org/keycloak/services/resources/ThemeResource.java]: Yep! Done!

File: services/src/main/java/org/keycloak/services/resources/ThemeResource.java (hunk 2 above)
Comment (rmartinc) [reply services/src/main/java/org/keycloak/services/resources/ThemeResource.java]: Done! It's more or less the same. :smile:

**Review by mposolda — APPROVED** [review]

**Review by jonkoops — APPROVED** [review]

**Pull request merged**